Sources/PackageCollections/PackageCollections.swift

```diff
@@ -0,0 +1,296 @@
+                       order: Int? = nil,
+                       to profile: PackageCollectionsModel.Profile? = nil,
+                       callback: @escaping (Result<PackageCollectionsModel.Collection, Error>) -> Void) {
+        let profile = profile ?? .default
+
+        if let errors = source.validate() {
+            return callback(.failure(MultipleErrors(errors)))
+        }
+
+        self.storage.collectionsProfiles.add(source: source, order: order, to: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success:
+                self.refreshPackageGroupFromSource(source: source, order: order, profile: profile, callback: callback)
+            }
+        }
+    }
+
+    func removeCollection(_ source: PackageCollectionsModel.CollectionSource,
+                          from profile: PackageCollectionsModel.Profile? = nil,
+                          callback: @escaping (Result<Void, Error>) -> Void) {
+        let profile = profile ?? .default
+
+        self.storage.collectionsProfiles.remove(source: source, from: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success:
+                self.storage.collectionsProfiles.exists(source: source, in: nil) { result in
+                    switch result {
+                    case .failure(let error):
+                        callback(.failure(error))
+                    case .success(let exists):
+                        if exists {
+                            callback(.success(()))
+                        } else {
+                            self.storage.collections.remove(identifier: .init(from: source), callback: callback)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    func moveCollection(_ source: PackageCollectionsModel.CollectionSource,
+                        to order: Int, in profile: PackageCollectionsModel.Profile? = nil,
+                        callback: @escaping (Result<Void, Error>) -> Void) {
+        let profile = profile ?? .default
+
+        self.storage.collectionsProfiles.move(source: source, to: order, in: profile, callback: callback)
+    }
+
+    func getCollection(_ source: PackageCollectionsModel.CollectionSource,
+                       callback: @escaping (Result<PackageCollectionsModel.Collection, Error>) -> Void) {
+        if let errors = source.validate() {
+            return callback(.failure(MultipleErrors(errors)))
+        }
+
+        self.storage.collections.get(identifier: .init(from: source)) { result in
+            switch result {
+            case .failure:
+                guard let provider = self.providers[source.type] else {
+                    return callback(.failure(UnknownProvider(source.type)))
+                }
+                provider.get(source, callback: callback)
```

Comment:
Let's add a comment to explain that we try to load package collection from local cache/storage first and only download via provider if not found.